Products/Zuul/facades/devicefacade.py

```diff
@@ -1011,9 +1011,12 @@ def getGraphDefinitionsForComponent(self, uid):
             components = list(getObjectsFromCatalog(obj.componentSearch, None, log))
 
         for component in components:
-            if graphDefs.get(component.meta_type):
-                continue
-            graphDefs[component.meta_type] = [graphDef.id for graphDef, _ in component.getGraphObjects()]
+            current_def = [graphDef.id for graphDef, _ in component.getGraphObjects()]
+            if component.meta_type in graphDefs:
+                prev_def = graphDefs[component.meta_type]
+                graphDefs[component.meta_type] = prev_def + list(set(current_def) - set(prev_def))
```

Comment:
This can be written with less complexity as:
```
    graphDefs[component.meta_type] = set(prev_def).union(current_def)
```
or
```
    graphDefs[component.meta_type] = set(prev_def) | set(current_def)
```